### PR TITLE
fix: GraphQL null argument parsing

### LINKED ISSUE #3012: Explicitly null GraphQL input values cause panic
We do not correctly handle explicitly null values in GQL requests.

### PR DESCRIPTION
**Pull request by nasdf**
## Relevant issue(s)

Resolves #3012

## Description

This PR fixes an issue where explicitly null GQL inputs would cause a panic.

## Tasks

- [x] I made sure the code is well commented, particularly hard-to-understand areas.
- [x] I made sure the repository-held documentation is changed accordingly.
- [x] I made sure the pull request title adheres to the conventional commit style (the subset used in the project can be found in [tools/configs/chglog/config.yml](tools/configs/chglog/config.yml)).
- [x] I made sure to discuss its limitations such as threats to validity, vulnerability to mistake and misuse, robustness to invalidation of assumptions, resource requirements, ...

## How has this been tested?

Added tests

Specify the platform(s) on which this was tested:
- MacOS

### CODE REVIEW

File: internal/planner/mapper/targetable.go [hunk 1]
```diff
@@ -110,7 +110,7 @@ func filterObjectToMap(mapping *core.DocumentMapping, obj map[connor.FilterKey]a
 	for k, v := range obj {
 		switch keyType := k.(type) {
 		case *PropertyIndex:
-			subObj := v.(map[connor.FilterKey]any)
+			subObj, _ := v.(map[connor.FilterKey]any)
```
Comment (AndrewSisley) [inline internal/planner/mapper/targetable.go:113]: todo: This looks like it was accidentally left in after debugging - can it be reverted?

File: tests/integration/mutation/create/with_null_input_test.go [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+			testUtils.SchemaUpdate{
+				Schema: `
+					type Users {
+						name: String
+					}
+				`,
+			},
+			testUtils.Request{
+				Request: `mutation {
+					create_Users(encrypt: null, input: {name: "Bob"}) {
```
Comment (AndrewSisley) [inline tests/integration/mutation/create/with_null_input_test.go:32]: question: `input` is not nillable? (I don't see any tests for that)

File: tests/integration/mutation/create/with_null_input_test.go [hunk 3]
```diff
@@ -0,0 +1,110 @@
+// Copyright 2024 Democratized Data Foundation
```
Comment (AndrewSisley) [inline tests/integration/mutation/create/with_null_input_test.go:1]: priase: Thanks for all theses tests Keenan, looks good :)

**Review by AndrewSisley — APPROVED** [review]
Looks good to me Keenan, thanks for sorting this out so quickly.  Approving now, but have a tiny todo and a question for you before merge please :)

File: internal/planner/mapper/targetable.go (hunk 1 above)
Comment (nasdf) [reply internal/planner/mapper/targetable.go:113]: This line was causing a panic when the property value is nil. Is there a different way I should be handling the nil value here?

Comment (AndrewSisley) [reply internal/planner/mapper/targetable.go:113]: Ah I see.  I would prefer a `if v != nil` wrapper then - it is a much stronger declaration of intent.

Comment (nasdf) [reply internal/planner/mapper/targetable.go]: done

File: tests/integration/mutation/create/with_null_input_test.go (hunk 2 above)
Comment (nasdf) [reply tests/integration/mutation/create/with_null_input_test.go:32]: nice catch! added a test

**Pull request merged**